### PR TITLE
unnecessary comment has removed from last commit, Project exercise part 1 complete, new implementation ensure delete …

### PR DESCRIPTION
**Pull request by cmabdullah**
Project exercise part 1 complete, new implementation ensure delete image while delete operation execute , almost Fix the Bug of Empty Contact Details , Contact Name * and Contact Group Name is validate till now.

### CODE REVIEW

File: grails-app/services/com/hmtmcse/ocb/ContactService.groovy [hunk 1]
```diff
@@ -62,7 +63,30 @@ class ContactService {
 
     def delete(Contact contact) {
         try {
-            contact.delete(flush: true)
```
Comment (hmtmcse) [inline grails-app/services/com/hmtmcse/ocb/ContactService.groovy]: The Implementation will work fine but it not proper implementation, you have to use FileUtil for this. and you are not allowed to use Holders everywhere. so the code is going to reject

**Comment by hmtmcse** [comment]
Please check the comments in codes

**Comment by cmabdullah** [comment]
Unnecessary comment has been removed from last commit, also improve code quality. Let me know if any correction is needed...

File: grails-app/services/com/hmtmcse/ocb/ContactService.groovy [hunk 2]
```diff
@@ -62,7 +63,20 @@ class ContactService {
...
+
+            def absPath = Holders.servletContext?.getRealPath("") + "contact-image/"
+
```
Comment (hmtmcse) [inline grails-app/services/com/hmtmcse/ocb/ContactService.groovy]: if you see this file
https://github.com/hmtmcse-com/grails-tutorial-contacts-book/blob/master/src/main/groovy/com/hmtmcse/ocb/FileUtil.groovy

Then you will see how the file uploadContactImage method, here you are seeing that for path it not using Holders, it calling methods from above, and you are seeing that, all file related work stored int FileUtil.groovy, you it recommended to bring your implementation to that file and call from controller. :)

**Review by hmtmcse — COMMENTED** [review]
Please see the code new comments

File: grails-app/services/com/hmtmcse/ocb/ContactService.groovy (hunk 2 above)
Comment (cmabdullah) [reply grails-app/services/com/hmtmcse/ocb/ContactService.groovy]: thanks for your valuable guidance, let me check update implementation.....

**Pull request closed without merging**